src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,143 @@
+package duke.util;
+
+import duke.command.AddCommand;
+import duke.command.InvalidCommand;
+import duke.command.Command;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.DeleteCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeArgumentException;
+import duke.exception.DukeDateTimeException;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class representing a Parser.
+ */
+public class Parser {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    /**
+     * Parses user input and returns a Command corresponding to user input.
+     *
+     * @param userInput The input of the user.
+     * @param tasks List of tasks.
+     * @return An executable command.
+     */
+    public static Command parse(String userInput, TaskList tasks) {
+        String[] split = userInput.split("\\s+");
+        String command = split[0];
+        Command res;
+        try {
+            switch (command) {

```

Comment:
Perhaps abstracting part of the code into methods with reasonable names is better in terms of readability?